test/index.test.js

```diff
@@ -1,9 +1,77 @@
 'use strict';
 
 const assert = require('chai').assert;
+const fs = require('fs');
+const path = require('path');
+
+/**
+ * Read a template file in the "data" directory by filename
+ * @method readConfig
+ * @param  {String}   filename Name of the file to read
+ * @return {Promise}           A promise that resolves to a string, which represents
+ *                             the contents of the file
+ */
+function readConfig(filename) {
+    const filepath = path.resolve(__dirname, 'data', filename);
+
+    return new Promise((resolve, reject) => {
```

Comment:
There are promise-based wrappers for `fs` (https://www.npmjs.com/package/mz). Or you could use the `Sync` version, `fs.readFileSync`, for tests to make this cleaner.